kubeflow/core/tensorboard.libsonnet

```diff
@@ -1,247 +1,153 @@
-  // params overrides s3params because params can be overwritten by the user to override the defaults.
-  s3params:: {
-    //  Name of the k8s secrets containing S3 credentials
-    s3SecretName: "",
-    // Name of the key in the k8s secret containing AWS_ACCESS_KEY_ID.
-    s3SecretAccesskeyidKeyName: "",
-
-    // Name of the key in the k8s secret containing AWS_SECRET_ACCESS_KEY.
-    s3SecretSecretaccesskeyKeyName: "",
-
-    // S3 region
-    s3AwsRegion: "us-west-1",
-
-    // TODO(jlewi): We should use util.toBool to automatically conver to actual boolean values.
-    // The use of strings is left over from when they were prototype parameters which only supports string type.
-
-    // true Whether or not to use https for S3 connections
-    s3UseHttps: "true",
-
-    // Whether or not to verify https certificates for S3 connections
-    s3VerifySsl: "true",
-
-    // URL for your s3-compatible endpoint.
-    s3Endpoint: "http://s3.us-west-1.amazonaws.com,",
-  } + $.params,
-
-
-  components:: {
-
-    all::
-      // TODO(jlewi): It would be better to structure s3 as a mixin.
-      // As an example it would be great to allow S3 and GCS parameters
-      // to be enabled simultaneously. This should be doable because
-      // each entails adding a set of environment variables and volumes
-      // to the containers. These volumes/environment variables shouldn't
-      // overlap so there's no reason we shouldn't be able to just add
-      // both modifications to the base container.
-      // I think we want to restructure things as mixins so they can just
-      // be added.
-      if $.params.s3Enable then
-        [
-          $.s3parts.tb,
-          $.s3parts.tfDeployment,
-        ]
-      else if $.params.cloud == "gcp" then
-        [
-          $.gcpParts.tb,
-          $.gcpParts.tfDeployment,
-        ]
-      else
-        [
-          $.parts.tb,
-          $.parts.tfDeployment,
-        ],
-  }.all,
-
-  parts:: {
-    // We define the containers one level beneath parts because combined with jsonnet late binding
-    // this makes it easy for users to override specific bits of the container.
-    tbContainer:: {
-      name: $.params.name,
-      image: $.params.defaultTbImage,
-      imagePullPolicy: "IfNotPresent",
-      args: [
-        $.params.logDir,
-        "--port=9000",
-      ],
-      command: [
-        "/usr/local/bin/tensorboard",
-      ],
-      ports: [
-        {
-          containerPort: 9000,
-        },
-      ],
-
-      resources: {
-        requests: {
-          memory: "1Gi",
-          cpu: "1",
-        },
-        limits: {
-          memory: "4Gi",
-          cpu: "4",
-        },
-      },
-    },  // tbContainer
-
-    tfDeployment: {
-      apiVersion: "extensions/v1beta1",
-      kind: "Deployment",
-      metadata: {
-        name: $.params.name,
-        namespace: $.params.namespace,
-        labels: $.params.labels,
+  local k = import "k.libsonnet",
```

Comment:
Should we move all of this into the .jsonnet file? This will make it easier for people to customize.

tensorboard is different from components like JupyterHub because end users will need to create many instances of tensorboard whereas most people won't be modifying JupyterHub.